streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java

```diff
@@ -580,6 +583,39 @@ public List<String> historyUploadJars() {
         .collect(Collectors.toList());
   }
 
+  @Override
+  public String k8sStartLog(Long id, Integer offset, Integer limit) throws Exception {
+    Application application = getById(id);
+    AssertUtils.state(application != null);
+    if (ExecutionMode.isKubernetesMode(application.getExecutionModeEnum())) {
+      CompletableFutureUtils.supplyTimeout(
+              CompletableFuture.supplyAsync(
+                  () ->
+                      KubernetesDeploymentHelper.watchDeploymentLog(
+                          application.getK8sNamespace(),
+                          application.getJobName(),
+                          application.getJobId())),
+              5,
+              TimeUnit.SECONDS,
+              success -> success,
+              exception -> {
+                String errorLog =
+                    String.format(
+                        "%s/%s_err.log", WebUtils.getAppTempDir(), application.getJobId());
+                File file = new File(errorLog);
+                if (file.exists()) {
+                  return errorLog;
+                } else {
+                  throw new ApiDetailException("get k8s job log failed: " + exception.getMessage());
+                }
+              })
```

Comment:
```suggestion

      CompletableFuture future = CompletableFuture.supplyAsync(
                  () ->
                      KubernetesDeploymentHelper.watchDeploymentLog(
                          application.getK8sNamespace(),
                          application.getJobName(),
                          application.getJobId()));
      CompletableFutureUtils.supplyTimeout(future,
              5,
              TimeUnit.SECONDS,
              success -> success,
              exception -> {
                String errorLog =
                    String.format(
                        "%s/%s_err.log", WebUtils.getAppTempDir(), application.getJobId());
                File file = new File(errorLog);
                if (file.exists()) {
                  return errorLog;
                } else {
                  throw new ApiDetailException("get k8s job log failed: " + exception.getMessage());
                }
                if (!future.isDone()) {
                  future.cancel(true)
                }
              })
```

The future need to be cancel after timeout.

Reply:
done